### ISSUE TITLE
x/vuln: Binary mode not compatible with golang 1.21.0

### ISSUE THREAD

**Issue by scottso** [golang/go#61936]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.21.0 linux/amd64
</pre>

### Does this issue reproduce at the latest version of golang.org/x/vuln?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=''
GOARCH='amd64'
GOBIN=''
GOCACHE='/root/.cache/go-build'
GOENV='/root/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMODCACHE='/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='linux'
GOPATH='/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/usr/local/go'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/usr/local/go/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='go1.21.0'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='gcc'
CXX='g++'
CGO_ENABLED='1'
GOMOD='/go/src/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build195800328=/tmp/go-build -gno-record-gcc-switches'
</pre></details>

### What did you do?
<pre>
$ govulncheck -mode=binary testme
</pre>

Any binary built with golang 1.21.0 will express this error.

### What did you expect to see?
```
Using govulncheck@v1.0.0 with vulnerability data from https://vuln.go.dev (last modified 2023-08-02 20:33:39 +0000 UTC).

Scanning your binary for known vulnerabilities...

No vulnerabilities found.

Share feedback at https://go.dev/s/govulncheck-feedback.
```

### What did you see instead?
```
Using govulncheck@v1.0.0 with vulnerability data from https://vuln.go.dev (last modified 2023-08-02 20:33:39 +0000 UTC).

Scanning your binary for known vulnerabilities...

govulncheck: could not parse provided binary: binary built using unsupported Go version: "go1.21.0"
```

**Comment by mknyszek** [comment]
CC @golang/vulndb

**Comment by zpavlinovic** [comment]
Could you confirm that this is still an issue with the latest version of govulncheck?

**Comment by corani** [comment]
I'm seeing the same issue for all my projects. Latest master seems to work:

```
Using govulncheck@v1.0.1-0.20230811173009-242ecf7a366e with vulnerability data from https://vuln.go.dev (last modified 2023-08-11 01:41:48 +0000 UTC).

Scanning your binary for known vulnerabilities...

No vulnerabilities found.
```

Could you create a new tagged release?

**Comment by corani** [comment]
Confirmed fixed in `v1.0.1`. Thanks for the quick turnaround!
